lib/p2p/Pool.ts

```diff
@@ -183,27 +183,42 @@ class Pool extends EventEmitter {
    * Attempt to create an outbound connection to a node using its known listening addresses.
    */
   private tryConnectNode = async (node: NodeConnectionInfo, retryConnecting = false) => {
+    if (!await this.tryConnectWithLastAddress(node, retryConnecting)) {
+      await this.tryConnectWithAdvertisedAddresses(node);
+    }
+  }
+
+  private tryConnectWithLastAddress = async (node: NodeConnectionInfo, retryConnecting = false) => {
+    const { lastAddress, nodePubKey } = node;
+
+    if (!lastAddress) return false;
+
+    try {
+      await this.addOutbound(lastAddress, nodePubKey, false);
+      return true;
+    } catch (err) {}
+
+    if (retryConnecting) {
+      try {
+        await this.addOutbound(lastAddress, nodePubKey, true);
```

Comment:
Don't we want a `return true;` after this line? It looks like it will always return false if we only manage to connect on retry. If so, this can actually be combined with the block above, just:

```typescript
try {
  await this.addOutbound(lastAddress, nodePubKey, retryConnecting);
  return true;
} catch (err) {}

return false;
```